server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -5471,9 +5472,14 @@ protected void validateNtwkOffDetails(final Map<Detail, String> details, final M
                 for (final PhysicalNetworkVO pNtwk : pNtwks) {
                     final List<String> pNtwkTag = pNtwk.getTags();
                     if (pNtwkTag == null || pNtwkTag.isEmpty()) {
-                        throw new CloudRuntimeException("Tags are not defined for physical network in the zone id=" + zoneId);
+                        if (!allowNullTag) {
+                            allowNullTag = true;
+                        } else {
+                            throw new CloudRuntimeException("There are more than 1 physical network with empty tag in the zone id=" + zoneId);
+                        }
+                    } else {
+                        pNtwkTags.addAll(pNtwkTag);
```

Comment:
method call

Reply:
you mean, move it to a method?

Reply:
yes @ravening, methods seem to grow out of hand in our project so any addition of more than one line I'd like to see in a separate method. As said I'm being pedantic, and so open for discussion, but as a matter of principle and keeping in mind that this is inherited code that will be passed on to posterity....